### PR TITLE
WIP: Run parallel testing with split-subset in Launchable

### PR DESCRIPTION
**Pull request by ono-max**

### CODE REVIEW

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -39,9 +51,56 @@ jobs:
...
+        if: ${{ matrix.test_task == 'check' || matrix.test_task == 'test-all' }}
+      - name: Set up Launchable
+        id: enable-launchable
+        uses: ./.github/actions/launchable/setup
+        with:
+          launchable-token: ${{ secrets.LAUNCHABLE_TOKEN }}
+          srcdir: src
+        if: ${{ matrix.test_task == 'check' || matrix.test_task == 'test-all' }}
+      - run: |
+          ruby -e 'File.write("launchable-subset-input.txt", Dir["test/**/*_test.rb", "test/**/test_*.rb"].map {|file| "file=#{file}"}.join("\n"))'
```
Comment (nobu) [inline .github/workflows/macos.yml:79]: `find` command is simpler for a simple task like this.

```suggestion
          find test -name "*_test.rb" -o -name "test_*.rb" -printf "file=%P\n" > launchable-subset-input.txt
```

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -86,12 +143,116 @@ jobs:
...
+        if: ${{ always() && (matrix.test_task == 'check' || matrix.test_task == 'test-all') }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+        if: ${{ always() && (matrix.test_task == 'check' || matrix.test_task == 'test-all') }}
+      - name: Install launchable command
+        run: |
+          pip install launchable
+          launchable split-subset --subset-id $(cat ../launchable-subset-id.txt) --bin 1/4 raw > launchable-subset.txt
+          echo "TESTS=$(cat launchable-subset.txt | tr '\n' ' ' | sed 's/file=/..\/src\//g')" >> $GITHUB_ENV
```
Comment (nobu) [inline .github/workflows/macos.yml:158]: `sed` applies the command for each line, and `$(...)` replaces `\n` with a space like `tr` does here.
```suggestion
          echo "TESTS=$(cat launchable-subset.txt | sed 's:file=:../src/:')" >> $GITHUB_ENV
```

**Review by nobu — COMMENTED** [review]
It doesn't feel nice to repeat all things four times.
And `skipped_tests` won't be repeated every time?

**Comment by ono-max** [comment]
@nobu 

Thank you for reviewing this PR. However, I'm going to close this PR because I don't feel that this solution is optimal. I apologize for any confusion.

**Pull request closed without merging**